pkg/verify/signed_entity.go

```diff
@@ -199,10 +202,22 @@ func WithCurrentTime() VerifierOption {
 	}
 }
 
+// WithNoObserverTimestamps configures the Verifier to not expect
+// any timestamps from either a Timestamp Authority or a Transparency Log
+// and to not use the current time to verify a certificate. This may only
+// be used when verifying with keys rather than certificates.
+func WithNoObserverTimestamps() VerifierOption {
+	return func(c *VerifierConfig) error {
+		c.allowNoTimestamp = true
+		return nil
+	}
+}
+
 func (c *VerifierConfig) Validate() error {
-	if !c.requireObserverTimestamps && !c.requireSignedTimestamps && !c.requireIntegratedTimestamps && !c.useCurrentTime {
+	if !c.requireObserverTimestamps && !c.requireSignedTimestamps && !c.requireIntegratedTimestamps && !c.useCurrentTime && !c.allowNoTimestamp {
 		return errors.New("when initializing a new Verifier, you must specify at least one of " +
-			"WithObserverTimestamps(), WithSignedTimestamps(), or WithIntegratedTimestamps()")
+			"WithObserverTimestamps(), WithSignedTimestamps(), or WithIntegratedTimestamps(). You may also specify " +
+			"WithCurrentTime() or WithNoObserverTimestamps()")
 	}
```

Comment:
Should we also add a check here to make sure that if you have `c.allowNoTimestamp` you don't have any of the other options specified?

Otherwise what should the behavior be if you do something like `c.requireObserverTimestamps` and `c.allowNoTimestamp`?

Reply:
Good idea, added this!